geo-traits/src/multi_line_string.rs

```diff
@@ -11,22 +11,16 @@ use geo_types::{CoordNum, LineString, MultiLineString};
 /// A MultiLineString is a collection of [`LineString`s][LineStringTrait].
 ///
 /// Refer to [geo_types::MultiLineString] for information about semantics and validity.
-pub trait MultiLineStringTrait: Sized {
-    /// The coordinate type of this geometry
-    type T;
-
+pub trait MultiLineStringTrait: Sized + GeometryTrait {
     /// The type of each underlying LineString, which implements [LineStringTrait]
-    type LineStringType<'a>: 'a + LineStringTrait<T = Self::T>
+    type InnerLineStringType<'a>: 'a + LineStringTrait<T = Self::T>
```

Comment:
Why this change? Is this just a rename?

Reply:
Since all these traits now extend `GeometryTrait`, they all have the associated types on `GeometryTrait`, including `LineStringType`. So option 1 is to reuse `GeometryTrait::LineStringType` for the inner type on a multi line string (and we can remove the associated type here), or option 2 is to do what I did here which is rename this occurrence and leave the parent associated types untouched and unused. What do you think?

Reply:
Oh I see, thanks for the explanation. I could imagine instances where the type for `LineString` is not the same as the inner type of a `MultiLineString`, so probably better to rename and keep them distinct?

Reply:
Yeah that's what I am thinking too